ethereum/jsonrpc/src/test/java/tech/pegasys/pantheon/ethereum/jsonrpc/internal/privacy/methods/eea/EeaGetTransactionReceiptTest.java

```diff
@@ -108,80 +106,103 @@
+    final TransactionLocation transactionLocation = new TransactionLocation(Hash.EMPTY, 0);
+    when(blockchain.getTransactionLocation(nullable(Hash.class)))
+        .thenReturn(Optional.of(transactionLocation));
+    final BlockBody blockBody =
+        new BlockBody(Collections.singletonList(transaction), Collections.emptyList());
+    when(blockchain.getBlockBody(any(Hash.class))).thenReturn(Optional.of(blockBody));
+    final BlockHeader mockBlockHeader = mock(BlockHeader.class);
+    when(blockchain.getBlockHeader(any(Hash.class))).thenReturn(Optional.of(mockBlockHeader));
+    when(mockBlockHeader.getNumber()).thenReturn(0L);
     final PrivateTransactionStorage privateTransactionStorage =
         mock(PrivateTransactionStorage.class);
-    final List<Transaction> mockListTx = Arrays.asList(mockTx, transaction);
-    final TransactionLocation transactionLocation = new TransactionLocation(mockBlockHash, 1);
-
-    doReturn(privateTransactionStorage).when(privacyParameters).getPrivateTransactionStorage();
-    when(privateTransactionStorage.getEvents(any(Bytes32.class)))
-        .thenReturn(Optional.of(mockLogList));
-    when(privateTransactionStorage.getOutput(any(Bytes32.class)))
-        .thenReturn(Optional.of(mockBytesValue));
+    when(privacyParameters.getPrivateTransactionStorage()).thenReturn(privateTransactionStorage);
+    when(privateTransactionStorage.getEvents(any(Bytes32.class))).thenReturn(Optional.empty());
+    when(privateTransactionStorage.getOutput(any(Bytes32.class))).thenReturn(Optional.empty());
+  }
 
+  @Test
+  public void returnReceiptIfTransactionExists() {
     final EeaGetTransactionReceipt eeaGetTransactionReceipt =
         new EeaGetTransactionReceipt(blockchainQueries, enclave, parameters, privacyParameters);
     final Object[] params = new Object[] {transaction.hash()};
     final JsonRpcRequest request = new JsonRpcRequest("1", "eea_getTransactionReceipt", params);
 
-    when(blockchainQueries.getBlockchain()).thenReturn(blockchain);
-    when(blockchain.getTransactionByHash(transaction.hash())).thenReturn(Optional.of(transaction));
-    when(blockchain.getTransactionLocation(nullable(Hash.class)))
-        .thenReturn(Optional.of(transactionLocation));
-    final BytesValueRLPOutput bvrlp = new BytesValueRLPOutput();
-    privateTransaction.writeTo(bvrlp);
-    when(enclave.receive(any(ReceiveRequest.class)))
-        .thenReturn(
-            new ReceiveResponse(
-                Base64.getEncoder().encodeToString(bvrlp.encoded().extractArray()).getBytes(UTF_8),
-                ""));
+    final JsonRpcSuccessResponse response =
+        (JsonRpcSuccessResponse) eeaGetTransactionReceipt.response(request);
+    final PrivateTransactionReceiptResult result =
+        (PrivateTransactionReceiptResult) response.getResult();
+
+    assertThat(result).isEqualToComparingFieldByField(expectedResult);
+  }
+
+  @Test
+  public void returnNullIfPMTDoesNotExist() {
```

Comment:
Maybe worth renaming - eg. enclaveThrowingResultsInSuccessButNullResponse?

Reply:
actually i have mixed up the method names here